src/React.Core/ReactIdGenerator.cs

```diff
@@ -0,0 +1,62 @@
+/*
+ *  Copyright (c) 2016-Present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant 
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+using System;
+using System.Threading;
+
+namespace React
+{
+	/// <summary>
+	/// Extension methods relating to GUIDs.
+	/// </summary>
+	public static class ReactIdGenerator
```

Comment:
Would be good to add an interface for this, and use the dependency injection framework to inject it into `ReactComponent`. That way, anyone can swap out the implementation if they want to (eg. if they want the old one back).

Reply:
(see `AssemblyRegistration.cs`, use `.AsSingleton()`)

Reply:
Nice idea, i can inject it to IReactEnvironment and create property IReactIdGenerator (like IBabel) and call it from constructor like "environment.ReactIdGenerator.Generate(...)"